### DISCUSSION TITLE
Nhost on google

### DISCUSSION THREAD

**Discussion by bmwasonga** [nhost/nhost#60] [Q&A] [answered]
I wanted to know how to add google authentication to my project. 

I have added the client Id and secret to my sign-in options on Nhost but I still end up getting  
''' 
Error 400: redirect_uri_mismatch
The redirect URI in the request, https://backend-bb872be2.nhost.app/auth/providers/google/callback, does not match the ones authorized for the OAuth client. To update the authorized redirect URIs, visit: https://console.developers.google.com/apis/credentials/oauthclient/684878247321-vn1fgd2mg58hvlne5gij94vdg73pn1to.apps.googleusercontent.com?project=684878247321
'''

is it something I might be doing wrong?

**Comment by elitan** [comment, marked as answer]
Yea I think you need add `https://backend-bb872be2.nhost.app/auth/providers/google/callback` to Google here:

![image](https://user-images.githubusercontent.com/331818/127271358-25e8782e-bb33-4685-84aa-7235081e01e2.png)

![image](https://user-images.githubusercontent.com/331818/127271382-ff83d040-66c1-4f2f-8980-fe74774cdcd5.png)

## Authorized JavaScript origins:

The HTTP origins that host your web application. This value can't contain wildcards or paths. If you use a port other than 80, you must specify it. For example: https://example.com:8080 

## Authorized redirect URIs (add your callback URL here)

Users will be redirected to this path after they have authenticated with Google. The path will be appended with the authorization code for access, and must have a protocol. It can’t contain URL fragments, relative paths, or wildcards, and can’t be a public IP address.

**Comment by bmwasonga** [comment]
this actually worked for me. 
Thank you

**Reply by elitan** [reply]
Fantastic! 🚀

**Comment by dionjwa** [comment]
This works when the Google OAuth app is in "test" mode but once you want to set to "production" and not limit users, the URL verification fails: https://github.com/nhost/nhost/discussions/983

**Reply by elitan** [reply]
How do you toggle a Google OAuth app from "test" to "production"? I didn't know about that setting.

**Reply by dionjwa** [reply]
<img width="726" alt="image" src="https://user-images.githubusercontent.com/260221/204726980-adbac6a3-f1e7-4683-8c95-75d89984bdd3.png">

**Reply by dionjwa** [reply]
The button"PUBLISH APP" in  `OAuth Consent screen`
